cosmos/providers/dbt/task_group.py

```diff
@@ -28,6 +28,9 @@ class DbtTaskGroup(CosmosTaskGroup):
         Defaults to "after_each"
     :param select: A dict of dbt selector arguments (i.e., {"tags": ["tag_1", "tag_2"]})
     :param exclude: A dict of dbt exclude arguments (i.e., {"tags": ["tag_1", "tag_2"]})
+    :param execution_mode: The execution mode in which the dbt project should be run.
+        Options are "local", "docker", and "kubernetes".
+        Defaults to "local"
```

Comment:
@juldrixx While testing, I realised that if the user passes something like:
```
    run_models = DbtTaskGroup(
        group_id="jaffle_shop",
        dbt_project_name="jaffle_shop",
        dbt_root_path="./dags/dbt/",
        conn_id="postgres_default",
        execution_mode="kubernetes",
        dbt_args={
            "schema": "public",
            "image": "dbt-jaffle-shop:0.0.6",
            "project_dir": PROJECT_DIR,
            "get_logs": True,
            "conn_id": "postgres_default",
            "image": "dbt-jaffle-shop:0.0.6",
            "is_delete_operator_pod": False,
        }
    )
```
The `project_dir` being set inside the inner DBT tasks, created by the task group, is not  `PROJECT_DIR`, but actually  `dbt_root_path` +  `dbt_project_name`. Whereas this a fair assumption for local execution, I don't think it is for the Dpckerfile - since at the moment, the user controls how the Dockerfile is created, and they may want to organise the files in a different way there. Could we adjust the code so that if the user gives `project_dir` in the `dbt_args`, it takes priority?

Reply:
I added a parameter `dbt_cmd_flags` in which you can put any parameter of the dbt command that will be used in it.
So it should be more like:

```python
    run_models = DbtTaskGroup(
        group_id="jaffle_shop",
        dbt_project_name="jaffle_shop",
        dbt_root_path="./dags/dbt/",
        conn_id="postgres_default",
        execution_mode="kubernetes",
        dbt_args={
            "schema": "public",
            "image": "dbt-jaffle-shop:0.0.6",
            "get_logs": True,
            "conn_id": "postgres_default",
            "image": "dbt-jaffle-shop:0.0.6",
            "is_delete_operator_pod": False,
            "dbt_cmd_flags": {                
                "project_dir": PROJECT_DIR,
            }
        }
    )
```

([here](https://github.com/astronomer/astronomer-cosmos/pull/194/files#diff-568541a54f2c6b233d209889aba3b3199918a4c06eac3d722fe4ae805cb42962R184))

Reply:
@juldrixx I found it a bit confusing `dbt_cmd_flags` inside `dbt_cmd_flags`. What do you think of changing to:
```
  run_models = DbtTaskGroup(
        group_id="jaffle_shop",
        dbt_project_name="jaffle_shop",
        dbt_root_path="./dags/dbt/",
        conn_id="postgres_default",
        execution_mode="kubernetes",
        operator_args={
            "image": "dbt-jaffle-shop:0.0.6",
            "get_logs": True,
            "is_delete_operator_pod": False,
        },
        dbt_args={
            "schema": "public",
            "project_dir": PROJECT_DIR,
        }
    )
```

If you're up for it, please, go ahead with the change and I'll approve the PR!

Reply:
I made the change, I'll let you confirm it if it's the way you want it. For now, you can put the parameters in `operator_args` or in `dbt_args`, it will make no difference.

Reply:
Thank you very much, @juldrixx , it all works!